source/uri-options/tests/connection-options.yml

```diff
@@ -0,0 +1,76 @@
+tests:
+    -
+        description: "Valid connection and timeout options are parsed correctly"
+        uri: "mongodb://example.com/?appname=URI-OPTIONS-SPEC-TEST&connectTimeoutMS=20000&heartbeatFrequencyMS=5000&localThresholdMS=3000&replicaSet=uri-options-spec&retryWrites=true&serverSelectionTimeoutMS=15000&socketTimeoutMS=7500"
+        valid: true
+        warning: false
+        hosts: ~
+        auth:
+            username: ~
+            password: ~
+            db: ~
+        options:
+            appname: "URI-OPTIONS-SPEC-TEST"
+            connectTimeoutMS: "20000"
```

Comment:
This was an intentional decision, although it could have gone either way. Basically, my view is that it's better to assume the minimum amount of work drivers might do here rather than the maximum; from the standpoint of the spec, all that matters is that correct values are accepted and incorrect values are rejected, not that the conversions are done immediately.

If the consensus is that it would be better to put in the actual types instead of strings, I'm fine with updating all the tests to do that.

Reply:
I vote for using correct types in the tests, since we already have prior art for doing so. I assume it will not cause problems for other drivers, and some could even take advantage of correct types in the test format to assert that they convert types correctly (I imagine some are already doing that in the connection string spec runner).

Reply:
Okay, I'll update them to use proper types

Reply:
Implemented